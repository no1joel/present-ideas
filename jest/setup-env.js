// test/setup-env.js
// add this to your setupFilesAfterEnv config in jest so it's imported for every test file
import fetch, { Headers, Request, Response } from "node-fetch";
import { TextDecoder as UtilTextDecoder } from "util";
import { server } from "../mock_server";

if (!globalThis.fetch) {
  // @ts-ignore
  globalThis.fetch = fetch;
  globalThis.Headers = Headers;
  // @ts-ignore
  globalThis.Request = Request;
  // @ts-ignore
  globalThis.Response = Response;
}

if (!globalThis.TextDecoder) {
  globalThis.TextDecoder = UtilTextDecoder;
}

beforeAll(() => server.listen());
// if you need to add a handler after calling setupServer for some specific test
// this will remove that handler for the rest of them
// (which is important for test isolation):
afterEach(() => server.resetHandlers());
afterAll(() => server.close());
