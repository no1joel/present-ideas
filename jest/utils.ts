import {
  render as testingLibraryRender,
  RenderOptions,
} from "@testing-library/vue";
import linkify from "vue-linkify";
import { createMemoryHistory, createRouter } from "vue-router";
import routes from "../present_ideas/present_ideas/static/router/routes";
import getStore from "../present_ideas/present_ideas/static/store";

const mockRouter = createRouter({ routes, history: createMemoryHistory() });

export const render = async (component, options = {}) => {
  const store = getStore();
  const defaultRenderOptions: RenderOptions = {
    global: {
      plugins: [store, mockRouter],
      directives: { linkified: linkify },
    },
  };
  const renderOptionsWithDefaults = { ...defaultRenderOptions, ...options };

  const result = testingLibraryRender(component, renderOptionsWithDefaults);

  return result;
};
