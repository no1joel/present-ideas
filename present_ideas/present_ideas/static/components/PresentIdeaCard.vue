<template>
  <div
    v-bind:class="[
      'card',
      { 'bg-secondary text-white': showClaimed && Claimed },
    ]"
  >
    <div class="card-header">
      <h5 class="card-title m-0" v-linkified>
        {{ Thing }}
      </h5>
      <p v-if="Price" class="card-text text-right text-muted">
        {{ Price }}
      </p>
      <p class="card-text text-right">
        <a href="#delete" class="btn btn-sm btn-outline-danger"> Delete </a>
      </p>
    </div>
    <div v-if="Notes" class="card-body">
      <p class="card-text" v-linkified>
        {{ Notes }}
      </p>
    </div>
    <div v-if="showClaimed" class="card-footer">
      <p v-if="Claimed">Claimed by <em>haven't done this bit yet!</em></p>
      <a href="#unclaim"> Clear? </a>
      <a href="#claim" class="btn btn-block btn-primary">Claim!</a>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    Thing: {
      type: String,
    },
    Notes: {
      type: String,
    },
    Price: {
      type: [String, Number],
    },
    showClaimed: {
      type: Boolean,
    },
    Claimed: {
      type: [String],
    },
  },
};
</script>