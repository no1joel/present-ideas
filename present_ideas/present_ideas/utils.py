from typing import Iterable, List

from sheets.spreadsheet import (
    add_row,
    delete_row,
    get_rows,
    get_worksheets,
    set_cell_value,
)


def check_name(name: str) -> bool:
    """Check if the provided name is a sheet."""

    worksheet_names = get_all_names()
    return name.lower() in worksheet_names


def get_all_names() -> Iterable[str]:
    """Return people's names."""

    return (
        worksheet.title.lower()
        for worksheet in get_worksheets()
        if "template" not in worksheet.title.lower()
    )


def get_present_ideas(their_name: str) -> List[dict]:
    # TODO: Embed urls
    _header, *rows = get_rows(their_name)
    presents = [{**row, "index": index} for index, row in enumerate(rows)]
    return presents


def get_row_index(thing_index: int) -> int:
    # TODO: Why is it plus 3?!
    return thing_index + 3
