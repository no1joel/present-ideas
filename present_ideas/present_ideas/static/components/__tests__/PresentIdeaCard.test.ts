import { render } from "../../../../../jest/utils";
import PresentIdeaCard from "../PresentIdeaCard.vue";

describe("PresentIdeaCard", () => {
  it("matches snapshot", async () => {
    const result = await render(PresentIdeaCard);
    expect(result.container).toMatchSnapshot();
  });
  it.todo("shows loading");
  it.todo("shows the right title");
  it.todo("shows the right notes");
  it.todo("shows the right price");
  it.todo("shows unclaimed if unclaimed");
  it.todo("shows who claimed it");
  it.todo("doesn't show claimed it if it's for you");
  it.todo("shows who added it");
  it.todo("lets you delete it if you added it");
  it.todo("lets you claim it if it's not yours");
});
