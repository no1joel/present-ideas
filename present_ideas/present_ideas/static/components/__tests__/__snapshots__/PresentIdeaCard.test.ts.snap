// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeoplePicker matches snapshot 1`] = `
<div>
  <select
    class="custom-select"
  >
    <!---->
     
    <option
      disabled="disabled"
      value=""
    >
      
    Pick a name!
  
    </option>
     
  </select>
</div>
`;

exports[`PresentIdeaCard matches snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <!---->
     
    <div
      class="card-header"
    >
      <h5
        class="card-title m-0"
      >
        
      
    
      </h5>
       
      <small>
        <!---->
         
        <!---->
      </small>
       
      <p
        class="card-text text-right mt-1"
      >
        <span>
          <transition-stub
            name=""
          >
            <span
              class=""
              style="display: none;"
            >
              <div
                class="popper popover"
              >
                <div
                  class="popover-header"
                >
                  Really delete?
                </div>
                 
                <div
                  class="popover-body"
                >
                  <button
                    class="btn btn-block btn-danger"
                    type="button"
                  >
                    
              Yep!
            
                  </button>
                </div>
              </div>
               
            </span>
          </transition-stub>
           
          <button
            class="btn btn-sm btn-outline-danger"
            type="button"
          >
            
          Delete
        
          </button>
        </span>
      </p>
    </div>
     
    <!---->
     
    <!---->
  </div>
</div>
`;
